frontend/cypress/integration/common/workload_logs.ts

```diff
@@ -0,0 +1,122 @@
+import { Given, Then, When } from "cypress-cucumber-preprocessor/steps";
+
+function goToWorkloadDetailPage(namespace: string, workload: string) {
+    // Go to workloads list
+    cy.get("#Workloads").click();
+
+    // Select "bookinfo" namespace
+    cy.get("#namespace-selector").click();
+    cy.get("#bulk-select-id").uncheck();
+    cy.get(`#namespace-list-item\\[${namespace}\\] input`).check();
+    cy.get("#namespace-selector").click();
+
+    // Open workload detail page
+    cy.get(`[data-test=VirtualItem_Nsbookinfo_${workload}] a`).click();
+}
+
+Given("I am on the {string} workload detail page of the {string} namespace", (workload, namespace) => {
+    goToWorkloadDetailPage(namespace, workload);
+});
+
+Given("I am on the logs tab of the {string} workload detail page of the {string} namespace", (workload, namespace) => {
+    goToWorkloadDetailPage(namespace, workload);
+    cy.get("[data-test=workload-details-logs-tab]").click();
+});
+
+When("I go to the Logs tab of the workload detail page", () => {
+    cy.get("[data-test=workload-details-logs-tab]").click();
+});
+
+When("I type {string} on the Show text field", showText => {
+    cy.get('#log_show').type(showText + '{enter}');
+});
+
+When("I type {string} on the Hide text field", showText => {
+    cy.get('#log_hide').type(showText + '{enter}');
+});
+
+When("I choose to show 10 lines of logs", () => {
+    cy.get('#wpl_tailLines-toggle').click();
+    cy.get('#10-0').click();
+});
+
+When("I select only the {string} container", containerName => {
+    let containerListForm = cy.get("[data-test=workload-logs-pod-containers]");
+    containerListForm.get('[type=checkbox]:checked').uncheck();
+
+    let podLabel = containerListForm.get("label").contains(containerName);
+    podLabel.invoke('attr', 'for')
+        .then(labelFor => {
+            cy.get(`#${labelFor}`).check();
+        });
+});
+
+When("I enable visualization of spans", function () {
+    cy.get("#spans-show-").check();
+});
+
+Then("I should see the {string} container listed", (containerName) => {
+    let containerListForm = cy.get("[data-test=workload-logs-pod-containers]");
+
+    let podLabel = containerListForm.get("label").contains(containerName);
+    podLabel.should('have.text', containerName);
+});
+
+Then("the {string} container should be checked", containerName => {
+    let containerListForm = cy.get("[data-test=workload-logs-pod-containers]");
+    let podLabel = containerListForm.get("label").contains(containerName);
+
+    podLabel.invoke('attr', 'for')
+        .then(labelFor => {
+            cy.get(`#${labelFor}`).should('be.checked');
+        });
+});
+
+Then("I should see some {string} pod selected in the pod selector", podNamePrefix => {
+    cy.get('#wpl_pods-toggle').should('include.text', podNamePrefix);
+});
+
+Then("the log pane should only show log lines containing {string}", filterText => {
+    let logPane = cy.get('#logsText');
```

Comment:
I _do_ want to assert that each paragraph (i.e. each line in the log pane) has `filterText`.

I think if I do `cy.get('#logsText').should('contain.text', filterText);` it will be enough if one paragraph (i.e. one line in the log pane) contains `filterText` for the assertion to pass. But I want to check all log lines.

I also could try `cy.get('#logsText').get('p').should('contain.text', filterText);`. I incidentally tried this on the `Then("the log pane should show spans")` (it's at the very bottom). It turns out that, despite the `get('p')` part would match multiple elements, the `should` won't check all matched elements :-/